### ISSUE TITLE
Support `nerdctl logs` for the containers running without -d

### ISSUE THREAD

**Issue by ningziwen** [containerd/nerdctl#1657]
### What is the problem you're trying to solve

Support `nerdctl logs` for the containers running without -d

[README.md](https://github.com/containerd/nerdctl/blob/main/README.md?plain=1#L682) already stated it is not supported yet. So opening a feature request instead of a bug report.

Reproducd by building source code from main branch.

```
$ sudo ./_output/nerdctl run -it ubuntu
# open a new terminal

$ sudo ./output/nerdctl ps # get the container id

$ sudo ./_output/nerdctl logs ccc97467def7
FATA[0000] failed to load logging config: failed to read log config file "/var/lib/nerdctl/1935db59/containers/default/ccc97467def7dd95330c6e872917e130180a1de2e4d6c34bdebeb16af434a5a6/log-config.json": open /var/lib/nerdctl/1935db59/containers/default/ccc97467def7dd95330c6e872917e130180a1de2e4d6c34bdebeb16af434a5a6/log-config.json: no such file or directory

```

### Describe the solution you'd like

Add the feature to support logs command with the container running without -d.

### Additional context

https://github.com/runfinch/finch/issues/85

**Comment by ningziwen** [comment]
Trying error log tracing in `nerdctl logs` command.

[nerdctl logs](https://github.com/containerd/nerdctl/blob/32fae2da2075f678579ae882a11ff6c32c537a67/cmd/nerdctl/logs.go#L152) -> [log_viewer.InitContainerLogViewer()](https://github.com/containerd/nerdctl/blob/32fae2da2075f678579ae882a11ff6c32c537a67/pkg/logging/log_viewer.go#L118) -> [logging.LoadLogConfig()](https://github.com/containerd/nerdctl/blob/32fae2da2075f678579ae882a11ff6c32c537a67/pkg/logging/logging.go#L132)

The error is failed in `os.ReadFile(logConfigFilePath)` without `no such file or directory`

The [logConfigFilePath](https://github.com/containerd/nerdctl/blob/32fae2da2075f678579ae882a11ff6c32c537a67/pkg/logging/logging.go#L122) is the log-config.json in current container's dataStore dir.

Next step: Find out where `nerdctl run -d` creates log-config.json and why `nerdctl run` without -d doesn't create it.

**Comment by ningziwen** [comment]
Just found this recent merged [PR](https://github.com/containerd/nerdctl/pull/1642) fixed the error. The log-config.json appears in the container ran by `nerdctl run -it` now.

However, there is no log in the container ran by `nerdctl run -it`.

Ran `while true; do $(echo date); sleep 1; done` in terminal of -it container. Then `nerdctl logs` returns empty.

Ran `sudo ./_output/nerdctl -d ubuntu sh -c "while true; do $(echo date); sleep 1; done"`, then `nerdctl logs` returns printed date lines.

**Comment by yzxiu** [comment]
Hi, 
When the container is run with `-it`，log data is output to nerdctl process stdout.
If want to get logs through `kubectl logs`, it may be necessary to copy the log to a log file(like: /var/lib/nerdctl/1935db59/containers/default/....-json.log) in another goroutine.

**Comment by ningziwen** [comment]
Currently, -it uses the [ioCreator](https://github.com/containerd/nerdctl/blob/2735dfc350a25f1857df8ca672723591a4cb4424/pkg/taskutil/taskutil.go#L84) to write stdOut to [console.Current()](https://github.com/containerd/nerdctl/blob/2735dfc350a25f1857df8ca672723591a4cb4424/cmd/nerdctl/run.go#L326).

-d uses the [ioCreator](https://github.com/containerd/nerdctl/blob/2735dfc350a25f1857df8ca672723591a4cb4424/pkg/taskutil/taskutil.go#L91) to write to [uri.String()](https://github.com/containerd/containerd/blob/46b706ac550c4d5a165e280df9219349f5f30e0c/cio/io.go#L237).

**Comment by AkihiroSuda** [comment]
@ningziwen Do you still plan to work on this?

**Comment by ningziwen** [comment]
> @ningziwen Do you still plan to work on this?

Yes but it will take some time. Feel free to take over if you have urgent need of this feature.

**Comment by danishprakash** [comment]
@ningziwen I can work on this if it's okay.

**Comment by ningziwen** [comment]
@danishprakash Sure go ahead.

**Comment by danishprakash** [comment]
I spent some time looking at this and based on what @ningziwen has already mentioned, I tried to create a multiwriter and create a ioCreator out of it but it's messy and didn't work primarily because `uri.String()` dir tree isn't setup yet or isn't the right path at the point of execution in `NewTask`. 

@yzxiu when you say we need to copy the logs over explicitly, shouldn't that piece go into containerd? I'm new to this project so still trying to understand the demarcations. Sorry If I missed anything.

**Comment by fahedouch** [comment]
I tried to fix this [here](https://github.com/containerd/nerdctl/pull/1947)   but stucked  because of [this](https://cloud-native.slack.com/archives/CGEQHPYF4/p1675108580366739?thread_ts=1675108580.366739&cid=CGEQHPYF4)

It looks like containerd io limitation

**Comment by danishprakash** [comment]
I think this issue can be closed then in lieu of #1946

**Comment by AkihiroSuda** [comment]
> I think this issue can be closed then in lieu of #1946

No, that one is a different topic

**Comment by vsiravar** [comment]
As per [SCOPE.md](https://github.com/containerd/containerd/blob/main/SCOPE.md#scope) which states that “Logging can be build on top of containerd because the container’s STDIO will be provided to the clients and they can persist any way they see fit. There is no io copying of container STDIO in containerd.” So this rules out modifying/creating the `NewTask(context.Context, cio.Creator, ...NewTaskOpts) (Task, error)` interface in containerd.This needs to be handled on the client side(i.e nerdctl) similar to what is done by [cri plugin](https://github.com/containerd/containerd/blob/87bf39a7f5580a86df739a787ced9664d1dc11bd/pkg/cri/io/container_io.go#L38). 

A proposal to solve this is for nerdctl to implement it's own `cio.Creator` which can  start the logging binary process as and send stdout and stederr streams to it using pipes, i.e mimic the [behavior of shim logging ](https://github.com/containerd/containerd/blob/87bf39a7f5580a86df739a787ced9664d1dc11bd/cmd/containerd-shim-runc-v2/process/io.go#L247) in the `cio.Creator` implementation. 

![image](https://github.com/containerd/nerdctl/assets/69735799/d7519e39-8f32-45e5-a1e2-c6e8e47b9542)

I made draft PR: https://github.com/containerd/nerdctl/pull/2683 and ran some sanity tests

```
 sudo nerdctl run --name interactive alpine sh -c "echo hello"
hello
$ sudo nerdctl logs interactive
hello
```
﻿
```
$ sudo nerdctl run -it --name itty alpine 
/ # ls
bin    dev    etc    home   lib    media  mnt    opt    proc   root   run    sbin   srv    sys    tmp    usr    var
/ # pwd
/
/ # exit
$ sudo nerdctl logs itty
/ # ls
bin    dev    etc    home   lib    media  mnt    opt    proc   root   run    sbin   srv    sys    tmp    usr    var
/ # pwd
/
/ # exit
```

cc @fahedouch @AkihiroSuda Let me know if this approach is okay and I can work on getting my PR merged. Thanks!
